### PR TITLE
metainfo: Fix listing paths

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: pkg/metainfo/kvmetainfo/objects.go [hunk 1]
```diff
@@ -108,7 +108,12 @@ func (db *DB) ListObjects(ctx context.Context, bucket string, options storj.List
 		return storj.ObjectList{}, errClass.New("invalid direction %d", options.Direction)
 	}
 
-	items, more, err := objects.List(ctx, options.Prefix, startAfter, endBefore, options.Recursive, options.Limit, meta.All)
+	// TODO: remove this hack-fix of specifying the last key
+	if options.Cursor == "" && (options.Direction == storj.Before || options.Direction == storj.Backward) {
+		endBefore = "\x7f\x7f\x7f\x7f\x7f\x7f\x7f"
+	}
+
+	items, more, err := db.objects.List(ctx, bucket+"/"+options.Prefix, startAfter, endBefore, options.Recursive, options.Limit, meta.All)
```
Comment (kaloyan-raev) [inline pkg/metainfo/kvmetainfo/objects.go]: This line should not be changed. The `object.Store` is obtained via `buckets.GetObjectStore()`. Hence there is also no need to prepend the bucket name to the prefix as this is done internally by the object store returned by `buckets.GetObjectStore()`.

Comment (egonelbre) [reply pkg/metainfo/kvmetainfo/objects.go]: Ah, right, missed the change above.

Comment (kaloyan-raev) [reply pkg/metainfo/kvmetainfo/objects.go]: `buckets.GetObjectStore()` takes a `bucket` param and returns the object store for the specific bucket. What is it missing for this use case?

Comment (egonelbre) [reply pkg/metainfo/kvmetainfo/objects.go]: Yup, fixed, my mistake.

**Pull request merged**